packages/opentelemetry-types/src/trace/TracerFactory.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tracer } from './tracer';
+
+/**
+ * TracerFactory is an object that creates a new {@link Tracer}.
+ */
+export interface TracerFactory {
```

Comment:
As per my understanding, `SpanProcessor` must move up to the `TracerFactory`. But `SpanProcessor` is SDK level concept, then this interface needs to move to the SDK / `Base` package, right?

Reply:
addSpanProcessor and other basictracer configuration would move into the BasicTracerFactory, which doesn't need to be reflected in the TracerFactory interface the way (BasicTracer).addSpanProcessor isn't reflected in the Tracer interface.

Reply:
Hmm, I see. Do you plan to implement that in the separate PR?

Reply:
@mayurkale22 yep, wanted to get feedback first since this is a fairly large change